### ISSUE TITLE
Query Job does not show the error traceback when it fails

### ISSUE THREAD

**Issue by zhquan** [chaoss/grimoirelab-sortinghat#487]
After executing `unify` I discover that I can't see the error when using the query `Job`

What I see is that the job failed but I don't know the reason
```
query {
	job(jobId: "652bed8d-3cf6-4a18-916a-838ddec7c1df"){
    status
    errors
  }
}
----
{
  "data": {
    "job": {
      "status": "failed",
      "errors": null
    }
  }
}
```

I find the error in `rq` output
```
16:34:40 default: sortinghat.core.jobs.unify(SortingHatContext(user=<User: quan>, job_id=None), ['0c0660cc3c4aa581bf8349cdff17b99d24bfd4bb', '0cac4ef12631d5b0ef2fa27ef0972..., None, ['email-name']) (652bed8d-3cf6-4a18-916a-838ddec7c1df)
16:34:40 Traceback (most recent call last):
  File "/home/quan/venvs/grimoirelab-sortinghat_graphql/lib/python3.6/site-packages/rq/worker.py", line 975, in perform_job
    rv = job.perform()
  File "/home/quan/venvs/grimoirelab-sortinghat_graphql/lib/python3.6/site-packages/rq/job.py", line 696, in perform
    self._result = self._execute()
  File "/home/quan/venvs/grimoirelab-sortinghat_graphql/lib/python3.6/site-packages/rq/job.py", line 719, in _execute
    return self.func(*self.args, **self.kwargs)
  File "/home/quan/github/chaoss/grimoirelab-sortinghat/sortinghat/core/jobs.py", line 314, in unify
    for rec in engine.recommend('matches', source_uuids, target_uuids, criteria):
  File "/home/quan/github/chaoss/grimoirelab-sortinghat/sortinghat/core/recommendations/engine.py", line 86, in _generate_recommendations
    for rec in recommender(*args, **kwargs):
  File "/home/quan/github/chaoss/grimoirelab-sortinghat/sortinghat/core/recommendations/matching.py", line 96, in recommend_matches
    matched = _find_matches(input_set, target_set, criteria, verbose=verbose)
  File "/home/quan/github/chaoss/grimoirelab-sortinghat/sortinghat/core/recommendations/matching.py", line 197, in _find_matches
    cdf_x = _filter_criteria(df_x, c)
  File "/home/quan/github/chaoss/grimoirelab-sortinghat/sortinghat/core/recommendations/matching.py", line 144, in _filter_criteria
    cdf = df[cols]
  File "/home/quan/venvs/grimoirelab-sortinghat_graphql/lib/python3.6/site-packages/pandas/core/frame.py", line 2912, in __getitem__
    indexer = self.loc._get_listlike_indexer(key, axis=1, raise_missing=True)[1]
  File "/home/quan/venvs/grimoirelab-sortinghat_graphql/lib/python3.6/site-packages/pandas/core/indexing.py", line 1254, in _get_listlike_indexer
    self._validate_read_indexer(keyarr, indexer, axis, raise_missing=raise_missing)
  File "/home/quan/venvs/grimoirelab-sortinghat_graphql/lib/python3.6/site-packages/pandas/core/indexing.py", line 1304, in _validate_read_indexer
    raise KeyError(f"{not_found} not in index")
KeyError: "['email-name'] not in index"
Traceback (most recent call last):
  File "/home/quan/venvs/grimoirelab-sortinghat_graphql/lib/python3.6/site-packages/rq/worker.py", line 975, in perform_job
    rv = job.perform()
  File "/home/quan/venvs/grimoirelab-sortinghat_graphql/lib/python3.6/site-packages/rq/job.py", line 696, in perform
    self._result = self._execute()
  File "/home/quan/venvs/grimoirelab-sortinghat_graphql/lib/python3.6/site-packages/rq/job.py", line 719, in _execute
    return self.func(*self.args, **self.kwargs)
  File "/home/quan/github/chaoss/grimoirelab-sortinghat/sortinghat/core/jobs.py", line 314, in unify
    for rec in engine.recommend('matches', source_uuids, target_uuids, criteria):
  File "/home/quan/github/chaoss/grimoirelab-sortinghat/sortinghat/core/recommendations/engine.py", line 86, in _generate_recommendations
    for rec in recommender(*args, **kwargs):
  File "/home/quan/github/chaoss/grimoirelab-sortinghat/sortinghat/core/recommendations/matching.py", line 96, in recommend_matches
    matched = _find_matches(input_set, target_set, criteria, verbose=verbose)
  File "/home/quan/github/chaoss/grimoirelab-sortinghat/sortinghat/core/recommendations/matching.py", line 197, in _find_matches
    cdf_x = _filter_criteria(df_x, c)
  File "/home/quan/github/chaoss/grimoirelab-sortinghat/sortinghat/core/recommendations/matching.py", line 144, in _filter_criteria
    cdf = df[cols]
  File "/home/quan/venvs/grimoirelab-sortinghat_graphql/lib/python3.6/site-packages/pandas/core/frame.py", line 2912, in __getitem__
    indexer = self.loc._get_listlike_indexer(key, axis=1, raise_missing=True)[1]
  File "/home/quan/venvs/grimoirelab-sortinghat_graphql/lib/python3.6/site-packages/pandas/core/indexing.py", line 1254, in _get_listlike_indexer
    self._validate_read_indexer(keyarr, indexer, axis, raise_missing=raise_missing)
  File "/home/quan/venvs/grimoirelab-sortinghat_graphql/lib/python3.6/site-packages/pandas/core/indexing.py", line 1304, in _validate_read_indexer
    raise KeyError(f"{not_found} not in index")
KeyError: "['email-name'] not in index"
```

**Comment by sduenas** [comment]
This is a bug because the errors should be returned on that field.

**Comment by sduenas** [comment]
There's another bug related to this one. When an invalid criteria value is given, it raises this error. Bug #491 opened to solve this other bug.

**Comment by sduenas** [comment]
Fixed by #489 .
